src/Microsoft.ML.TensorFlow/TensorflowTransform.cs

```diff
@@ -17,125 +19,295 @@
+
+        public static int BatchSize = 1;
+        public const string Summary = "Transforms the data using the TensorFlow model.";
+        public const string UserName = "TensorFlowTransform";
+        public const string ShortName = "TFTransform";
+        public const string LoaderSignature = "TensorFlowTransform";
+
+        private static VersionInfo GetVersionInfo()
+        {
+            return new VersionInfo(
+                modelSignature: "TENSFLOW",
+                //verWrittenCur: 0x00010001, // Initial
+                verWrittenCur: 0x00010002,  // Upgraded when change for multiple outputs was implemented.
+                verReadableCur: 0x00010002,
+                verWeCanReadBack: 0x00010001,
+                loaderSignature: LoaderSignature);
+        }
+
+        /// <summary>
+        /// Convenience constructor for public facing API.
+        /// </summary>
+        /// <param name="env">Host Environment.</param>
+        /// <param name="input">Input <see cref="IDataView"/>. This is the output from previous transform or loader.</param>
+        /// <param name="modelFile">This is the frozen TensorFlow model file. https://www.tensorflow.org/mobile/prepare_models </param>
+        /// <param name="name">Name of the output column. Keep it same as in the TensorFlow model.</param>
+        /// <param name="source">Name of the input column(s). Keep it same as in the TensorFlow model.</param>
+        public static IDataTransform Create(IHostEnvironment env, IDataView input, string modelFile, string name, params string[] source)
+        {
+            env.CheckNonWhiteSpace(modelFile, nameof(modelFile));
+            env.CheckUserArg(File.Exists(modelFile), nameof(modelFile));
+            return new TensorFlowTransform(env, File.ReadAllBytes(modelFile), source, new[] { name }).MakeDataTransform(input);
+        }
+
+        /// <summary>
+        /// Convenience constructor for public facing API.
+        /// </summary>
+        /// <param name="env">Host Environment.</param>
+        /// <param name="input">Input <see cref="IDataView"/>. This is the output from previous transform or loader.</param>
+        /// <param name="modelFile">This is the frozen tensorflow model file. https://www.tensorflow.org/mobile/prepare_models </param>
+        /// <param name="names">Name of the output column(s). Keep it same as in the Tensorflow model.</param>
+        /// <param name="source">Name of the input column(s). Keep it same as in the Tensorflow model.</param>
+        public static IDataTransform Create(IHostEnvironment env, IDataView input, string modelFile, string[] names, string[] source)
+        {
+            env.CheckNonWhiteSpace(modelFile, nameof(modelFile));
+            env.CheckUserArg(File.Exists(modelFile), nameof(modelFile));
+            return new TensorFlowTransform(env, File.ReadAllBytes(modelFile), source, names).MakeDataTransform(input);
+        }
+
+        // Factory method for SignatureLoadModel.
```

Comment:
>// Fact [](start = 8, length = 7)

xml style documentation, since it is a public method

Reply:
OR we could make that stuff `internal` (and change `ComponentCatalog` to look for `internal`. 

---
In reply to: [215723539](https://github.com/dotnet/machinelearning/pull/840#discussion_r215723539) [](ancestors = 215723539)

Reply:
Maybe later though.

---
In reply to: [215728658](https://github.com/dotnet/machinelearning/pull/840#discussion_r215728658) [](ancestors = 215728658,215723539)

Reply:
It's public because of flaws in our dependency injection.

---
In reply to: [215728692](https://github.com/dotnet/machinelearning/pull/840#discussion_r215728692) [](ancestors = 215728692,215728658,215723539)